### ISSUE TITLE
`:ObsidianRename` fails with neovim 0.9.*

### ISSUE THREAD

**Issue by Zenith-the-realest** [epwalsh/obsidian.nvim#301]
### The bug:

attempting to use :ObsidianRename results in this error:

```
Error executing Lua callback: ...nvim/lazy/obsidian.nvim/lua/obsidian/commands/rename.lua:68: attempt to call field 'joinpath' (a nil value)                  
stack traceback:                                                                                                                                              
        ...nvim/lazy/obsidian.nvim/lua/obsidian/commands/rename.lua:68: in function 'func'                                                                    
        ...e/nvim/lazy/obsidian.nvim/lua/obsidian/commands/init.lua:63: in function <...e/nvim/lazy/obsidian.nvim/lua/obsidian/commands/init.lua:62>
```

### My `lazy-lock.json` file:
```json
{
  "Comment.nvim": { "branch": "master", "commit": "0236521ea582747b58869cb72f70ccfa967d2e89" },
  "LuaSnip": { "branch": "master", "commit": "8ae1dedd988eb56441b7858bd1e8554dfadaa46d" },
  "base46": { "branch": "v2.0", "commit": "3f2b658cbd6650ddaf2bae3233e143a41ca25b1a" },
  "cmp-buffer": { "branch": "main", "commit": "3022dbc9166796b644a841a02de8dd1cc1d311fa" },
  "cmp-nvim-lsp": { "branch": "main", "commit": "5af77f54de1b16c34b23cba810150689a3a90312" },
  "cmp-nvim-lua": { "branch": "main", "commit": "f12408bdb54c39c23e67cab726264c10db33ada8" },
  "cmp-path": { "branch": "main", "commit": "91ff86cd9c29299a64f968ebb45846c485725f23" },
  "cmp_luasnip": { "branch": "master", "commit": "05a9ab28b53f71d1aece421ef32fee2cb857a843" },
  "friendly-snippets": { "branch": "main", "commit": "53d3df271d031c405255e99410628c26a8f0d2b0" },
  "fzf": { "branch": "master", "commit": "20240101197348a78db5a1dac945a2dff6a4e654" },
  "fzf-lua": { "branch": "main", "commit": "f4f3671ebc89dd25a583a871db07529a3eff8b64" },
  "fzf.vim": { "branch": "master", "commit": "1e054c1d075d87903647db9320116d360eb8b024" },
  "gitsigns.nvim": { "branch": "main", "commit": "d195f0c35ced5174d3ecce1c4c8ebb3b5bc23fa9" },
  "indent-blankline.nvim": { "branch": "master", "commit": "b7aa0aed55887edfaece23f7b46ab22232fc8741" },
  "lazy.nvim": { "branch": "main", "commit": "96584866b9c5e998cbae300594d0ccfd0c464627" },
  "markdown-preview.nvim": { "branch": "master", "commit": "a923f5fc5ba36a3b17e289dc35dc17f66d0548ee" },
  "mason.nvim": { "branch": "main", "commit": "a09da6ac634926a299dd439da08bdb547a8ca011" },
  "mini.pick": { "branch": "main", "commit": "29ccad7b9facb790d016a07c88fb55da5280297b" },
  "nvim-autopairs": { "branch": "master", "commit": "9fd41181693dd4106b3e414a822bb6569924de81" },
  "nvim-cmp": { "branch": "main", "commit": "538e37ba87284942c1d76ed38dd497e54e65b891" },
  "nvim-colorizer.lua": { "branch": "master", "commit": "85855b38011114929f4058efc97af1059ab3e41d" },
  "nvim-lspconfig": { "branch": "master", "commit": "ce0e625df61be77abe1340fbc9afe9ad39b31dd8" },
  "nvim-notify": { "branch": "master", "commit": "1576123bff3bed67bc673a3076e591abfe5d8ca9" },
  "nvim-surround": { "branch": "main", "commit": "703ec63aa798e5e07d309b35e42def34bebe0174" },
  "nvim-tree.lua": { "branch": "master", "commit": "f1b3e6a7eb92da492bd693257367d9256839ed3d" },
  "nvim-treesitter": { "branch": "master", "commit": "8ae4080c5ba59d81b4e49525d4168807bc098e73" },
  "nvim-web-devicons": { "branch": "master", "commit": "3e24abe1ae66532135cec911562f553fe247cb56" },
  "nvterm": { "branch": "main", "commit": "3e43be1d0ca60cc5e2dfc2d289b06577e7e57e98" },
  "obsidian.nvim": { "branch": "main", "commit": "dbd81af68896a6276001dd9f65c7cb965b411338" },
  "plenary.nvim": { "branch": "master", "commit": "55d9fe89e33efd26f532ef20223e5f9430c8b0c0" },
  "pomo.nvim": { "branch": "main", "commit": "bd32d8a696c73d62fc7ccb9be8f8d533883520fc" },
  "telescope.nvim": { "branch": "master", "commit": "87e92ea31b2b61d45ad044cf7b2d9b66dad2a618" },
  "ui": { "branch": "v2.0", "commit": "1737a2a98e18b635480756e817564b60ff31fc53" },
  "vim-markdown": { "branch": "master", "commit": "46add6c3017d3e4035dc10ffa9cb54221d8dfe1a" },
  "which-key.nvim": { "branch": "main", "commit": "4433e5ec9a507e5097571ed55c02ea9658fb268a" }
}
```

### Versions

`nvim --version`:

```bash
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1702233742

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

`nvim --headless -c 'lua require("obsidian").info()' -c q`:

```
[obsidian.nvim (v2.5.3)] Commit SHA: dbd81af68896a6276001dd9f65c7cb965b411338
[plenary.nvim] Commit SHA: 55d9fe89e33efd26f532ef20223e5f9430c8b0c0
[nvim-cmp] Commit SHA: 538e37ba87284942c1d76ed38dd497e54e65b891
[fzf-lua] Commit SHA: f4f3671ebc89dd25a583a871db07529a3eff8b64
[fzf.vim] Commit SHA: 1e054c1d075d87903647db9320116d360eb8b024
[mini.pick] Commit SHA: 29ccad7b9facb790d016a07c88fb55da5280297b
[vim-markdown] Commit SHA: 46add6c3017d3e4035dc10ffa9cb54221d8dfe1a
ripgrep 14.0.3
```

**Comment by epwalsh** [comment]
Hey @Zenith-the-realest, it seems that `vim.fs.joinpath()` is a recent addition to the nvim Lua API and unfortunately I overlooked that. Upgrading neovim to nightly (if you want to risk it) will fix that.

**Comment by Zenith-the-realest** [comment]
This solved it! Thanks.

**Comment by kappq** [comment]
Is there a way to fix this without upgrading to nightly?

**Comment by epwalsh** [comment]
@kappq we'd have to write our own version of `vim.fs.joinpath`. If someone wants to do that I'd be happy to review the PR.

**Comment by epwalsh** [comment]
Fixed in #432.
